### PR TITLE
ci: Update

### LINKED ISSUE #1502: Update CI matrix
* gcc 11-13
* debian 12
* fairsoft nov22, jan24

### PR DESCRIPTION
**Pull request by dennisklein**
Resolves #1502

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -105,29 +109,23 @@ pipeline{
...
+            [os: 'debian', ver: '11', arch: 'x86_64', compiler: 'gcc-10', fairsoft: 'jan24_patches',    image: 'jan24'],
+            [os: 'debian', ver: '11', arch: 'x86_64', compiler: 'gcc-10', fairsoft: 'jan24_patches_mt', image: 'jan24'],
+            [os: 'debian', ver: '11', arch: 'x86_64', compiler: 'gcc-10', fairsoft: 'nov22_patches',    image: 'nov22'],
+            [os: 'debian', ver: '11', arch: 'x86_64', compiler: 'gcc-10', fairsoft: 'apr22_patches',    image: 'apr22'],
+            [os: 'debian', ver: '12', arch: 'x86_64', compiler: 'gcc-12', fairsoft: 'jan24_patches',    image: 'jan24'],
+            [os: 'debian', ver: '12', arch: 'x86_64', compiler: 'gcc-12', fairsoft: 'jan24_patches_mt', image: 'jan24'],
+            [os: 'debian', ver: '12', arch: 'x86_64', compiler: 'gcc-12', fairsoft: 'nov22_patches',    image: 'nov22'],
+            // [os: 'ubuntu',   ver: 'rolling', arch: 'x86_64', compiler: 'current', fairsoft: 'dev',
+            //                  check: 'warnings',
+            //                  extra: '-DUSE_CLANG_TIDY=ON -DBUILD_MBS=OFF -DBUILD_PROOF_SUPPORT=OFF'],
```
[the remark this answers is not in the document]
Comment (dennisklein) [reply Jenkinsfile:125]: Nothing, will re-enable, once all the other images are finished. I have not changed this image, so no need to build it during testing the others.

**Comment by ChristianTackeGSI** [comment]
If you create some scripts to setup the containers, can you please add/update them at [`/tests/ci/container/`](https://github.com/FairRootGroup/FairRoot/tree/dev/tests/ci/container)? Either in this PR or a dedicated one?

**Comment by dennisklein** [comment]
> If you create some scripts to setup the containers, can you please add/update them at [`/tests/ci/container/`](https://github.com/FairRootGroup/FairRoot/tree/dev/tests/ci/container?rgh-link-date=2024-04-02T17%3A37%3A45Z)? Either in this PR or a dedicated one?

The top-level steering was indeed once again manual. So, nothing sophisticated yet to write down.

**Review by ChristianTackeGSI — APPROVED** [review]
It looks like `ex_MQ_pixel_split` fails a lot of newer distributions?
Do you want to incorporate #1513 into this PR or merge it and rebase this PR?

That said: Things look good!

**Comment by dennisklein** [comment]
on ubuntu 22.04 we have a number of tests that output successful execution and then segfault on shutdown it seems. but ye, let's merge this and work on the errors/warnings in follow-up PRs.

**Comment by dennisklein** [comment]
Ah, and I only added the Geant4 multi-threaded variant on FairSoft `jan24` images intentionally to keep the amount of checks smaller.

**Pull request merged**